flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -159,7 +157,7 @@ public TaskManagerRunner(Configuration configuration, ResourceID resourceId) thr
 		this.terminationFuture = new CompletableFuture<>();
 		this.shutdown = false;
 
-		MemoryLogger.startIfConfigured(LOG, configuration, metricQueryServiceActorSystem);
+		MemoryLogger.startIfConfigured(LOG, configuration, ((AkkaRpcService) metricQueryServiceRpcService).getActorSystem());
```

Comment:
The existing code is a bit odd; it should've never passed the MQS actor system but that of the RPC service, since that one more closely resembles the process as a whole.

Reply:
In any case I rather dislike that the MemoryLogger knows it is dealing with an ActorSystem. This makes it necessary to break the `RpcService` abstraction here, and generally exposes way more than necessary.

We should only pass a `CompletableFuture<Void>` that finishes upon termination of the TM, like `terminationFuture`.
I've opened https://issues.apache.org/jira/browse/FLINK-12057 and will open a PR shortly, which I'd prefer to resolve first.